### ISSUE TITLE
IDE0051 false positive

### ISSUE THREAD

**Issue by rdeago** [dotnet/roslyn#68162]
**Version Used**: Visual Studio 2022 v17.5.5

**Steps to Reproduce**:

1. Have a `private const string` inside a type.
2. Use said constant _only_ as an attribute parameter.
3. Verify that you get IDE0051 at the line of the constant definition.

SharpLab has no IDE messages, so here's the code, ready to paste in a new Visual Studio project:

```C#
using System;

public class Program
{
    private const string MyName = "Paul";
    
    // IDE0051: Private member 'MyName' is unused
    [Test(Name = MyName)]
    public static void Main()
    {
    }
}

public class TestAttribute : Attribute
{
    public string? Name { get; set; }
}
```

**Diagnostic Id**: IDE0051

**Expected Behavior**: Using a private constant as an attribute parameter should be deemed _actually using_ the constant, therefore not triggering IDE0051.

**Actual Behavior**: Using a private constant only as an attribute parameter triggers IDE0051.

**Comment by mavasani** [comment]
Verified no repro on VS2022 17.7.5
